src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+                    Task tsk = ls.get(numToBeDeleted);
+                    ls.remove(numToBeDeleted);
+                    ui.showDeleteMsg(ls, tsk);
+                } catch (Exception ex) {
+                    ui.showDeleteError();
+                }
+                break;
+            case CHECK:
+                try {
+                    String checkDate = arr[1];
+                    LocalDate date = LocalDate.parse(checkDate);
+                    if (ls.isEmpty()) {
+                        ui.showCheckNoTask();
+                    } else {
+                        ui.horizontalDiv();
+                        int counter = 0;
+                        for (int i = 0; i < ls.size(); i++) {
+                            Task task = ls.get(i);
+                            if (task instanceof Event) {
+                                if (((Event) task).getAt().equals(date) && !task.isDone()) {
+                                    counter += 1;
+                                    System.out.println(counter + ". " + task.toString());
+                                }
+                            } else if (task instanceof Deadline && !task.isDone()) {
+                                if (((Deadline) task).getDate().equals(date)) {
+                                    counter += 1;
+                                    System.out.println(counter + ". " + task.toString());
+                                }
+                            }
+                        }
+                        ui.showCheckTask(counter, date);
+                    }
+                } catch (Exception ex) {
+                    ui.showCheckError();
+                }
+                break;
+            case FIND:
+                try {
+                    String keyword = arr[1];
+                    if (ls.isEmpty()) {
+                        ui.showCheckNoTask();
+                    } else {
+                        ui.horizontalDiv();
+                        int counter = 0;
+                        for (int i = 0; i < ls.size(); i++) {
+                            Task task = ls.get(i);
+                            if (task.getDescription().contains(keyword)) {
+                                counter++;
+                                System.out.println(counter + ". " + task.toString());
+                            }
+                        }
+                        ui.showFindTask(counter, keyword);
+                    }
+                } catch (Exception ex) {
+                    ui.showFindError();
+                }
+                break;
+            default:
+                ui.invalidInput();
+                break;
+            }
+        }
+    }
+
+    /**
+     * This is the main method which will create an instance of duke.Duke
+     * and call on the method run.
+     *
+     * @param args Unused.
+     * @throws IOException On input error.
+     */
+    public static void main(String[] args) throws IOException {
```

Comment:
One suggestion would be to make a parser class to take care of the logic as well as making full use of the UI class to scan in your input.